godmode.txt

```diff
@@ -0,0 +1,44 @@
+
+---How do we use GodMode?---
+
+GodMode can be activated in two different ways. 
+
+If we want to run the entire test suite using GodMode, then use the `GodMode All tests` option under the green `Run` button in the Eclipse toolbar
+If we want to execute a specific test file using GodMode, then update BaseUiTestCase class and set ```enableGodMode = true``` at the top of the class implementation. Please remember to set it back to false when done.
+
+---When do we use GodMode?---
+
+GodMode can be used to create a new source file for a new browser test. This is done by simply assuming that the "expected" source is available and writing the test code. Then by executing the method using GodMode, the "expected" source will be generated and saved with the name specified in the test code. The following code snippet illustrates this:
+
+The following code is written WITHOUT a studentHomeTypicalHTML.html file. Then when this test is executed using GodMode, studentHomeTypicalHTML.html is automatically generated with the expected source.
+
+...
+StudentHomePage studentHomePage = loginAdminToPage(browser, detailsPageUrl, StudentHomePage.class);
+studentHomePage.verifyHtmlMainContent("/studentHomeTypicalHTML.html");
+...
+
+GodMode can also be used to update test cases when there are EXPECTED changes in the source of the webpage. This is done by executing those test cases with GodMode enabled. At the end of the test run, the expected source is updated with the new changes. 
+
+
+---Best Practices---
+
+1. Ensure that GodMode is only used when necessary, that is when there are new test cases being created or when an update to the existing tests is foreseen.
+2. Please remember to disable GodMode once the necessary changes have been made, before committing the changes.
+3. Please confirm that all the changes made by GodMode are EXPECTED. If any unexpected changes are made, please ask for assistance in the issue tracker or create a new issue if need be.
+4. Run the test suite once without GodMode enabled, after all the necessary changes have been made.
+
+
+--- Final Note ---
+
+DO NOT create or modify the "expected" html pages in the tests manually. Use GodMode even for seemingly trivial changes. Also, note that the generated html may not reflect the browser's source identically. Some modifications have been made to achieve cross-compatibility (eg: white space standardization).
+
+In general, only the lines that are modified should be changed by GodMode. However, since GodMode standardizes the white spacing, sometimes multiple (seemingly unrelated) lines may be affected due to changes in the indentation. In such cases, the reviewer can use "?w=1" in the *File Changes* tab of a pull request to ignore all the white space changes.
```

Comment:
1. Agreed, I guess we dont have to repeat ourselves.
2. Ok, I will add some examples